MaterialDesignThemes.Wpf/NumericUpDown.cs

```diff
@@ -0,0 +1,257 @@
+                RoutedEvent = ValueChangedEvent
+            };
+            numericUpDown.RaiseEvent(args);
+            if (numericUpDown._textBoxField is { } textBox)
+            {
+                textBox.Text = ((int)e.NewValue).ToString(CultureInfo.CurrentUICulture);
+            }
+
+            if (numericUpDown._increaseButton is { } increaseButton)
+            {
+                increaseButton.IsEnabled = numericUpDown.Value != numericUpDown.Maximum;
+            }
+
+            if (numericUpDown._decreaseButton is { } decreaseButton)
+            {
+                decreaseButton.IsEnabled = numericUpDown.Value != numericUpDown.Minimum;
+            }
+        }
+    }
+
+    private static object CoerceNumericValue(DependencyObject d, object value)
+    {
+        if (d is NumericUpDown numericUpDown &&
+            value is int numericValue)
+        {
+            numericValue = Math.Min(numericUpDown.Maximum, numericValue);
+            numericValue = Math.Max(numericUpDown.Minimum, numericValue);
+            return numericValue;
+        }
+        return value;
+    }
+    #endregion ValueProperty
+
+    #region DependencyProperty : IncreaseCommandProperty
+    public ICommand? IncreaseCommand
+    {
+        get => (ICommand?)GetValue(IncreaseCommandProperty);
+        set => SetValue(IncreaseCommandProperty, value);
+    }
+    public static readonly DependencyProperty IncreaseCommandProperty =
+            DependencyProperty.Register(nameof(IncreaseCommand), typeof(ICommand), typeof(NumericUpDown), new PropertyMetadata(null));
+
+    public object? IncreaseCommandParameter
+    {
+        get => (object?)GetValue(IncreaseCommandParameterProperty);
+        set => SetValue(IncreaseCommandParameterProperty, value);
+    }
+    public static readonly DependencyProperty IncreaseCommandParameterProperty =
+        DependencyProperty.Register(nameof(IncreaseCommandParameter), typeof(object), typeof(NumericUpDown), new PropertyMetadata(null));
+
+
+    #endregion DependencyProperty : IncreaseCommandProperty
+
+    #region DependencyProperty : DecreaseCommandProperty
+    public ICommand? DecreaseCommand
+    {
+        get => (ICommand?)GetValue(DecreaseCommandProperty);
+        set => SetValue(DecreaseCommandProperty, value);
+    }
+    public static readonly DependencyProperty DecreaseCommandProperty =
+            DependencyProperty.Register(nameof(DecreaseCommand), typeof(ICommand), typeof(NumericUpDown), new PropertyMetadata(default(ICommand?)));
+
+    public object? DecreaseCommandParameter
+    {
+        get => (object?)GetValue(DecreaseCommandParameterProperty);
+        set => SetValue(DecreaseCommandParameterProperty, value);
+    }
+
+    public static readonly DependencyProperty DecreaseCommandParameterProperty =
+        DependencyProperty.Register(nameof(DecreaseCommandParameter), typeof(object), typeof(NumericUpDown), new PropertyMetadata(null));
+
+    #endregion DependencyProperty : DecreaseCommandProperty
```

Comment:
Let me know if I am completely off the rails here, but here are my thoughts.

I think these commands are simply left-overs from when the XAML was using them. And I agree that is not common practice for a custom control; you should rely on the template parts and deal with this stuff in the code-behind.

Also, in their current state, I really don't think they should be part of the public API for a `NumericUpDown`. If you look at a [Slider](https://source.dot.net/#PresentationFramework/System/Windows/Controls/Slider.cs,96) for example, it does expose some static `RoutedCommands` to modify the value, but they don't have public setters. This way the calling code is not involved in determining whether `Command.CanExecute` returns true/false which would be a really weird thing; it is the control itself that has the knowledge to determine this, not the caller. The commands are still exposed so they can be bound to and thus executed in custom UI/flows.